From 96569a687f92e0d2deb73b95c445e568ef785175 Mon Sep 17 00:00:00 2001
From: Timothy D Witham <twitham@sbcglobal.net>
Date: Mon, 21 Aug 2023 23:50:51 -0500
Subject: [PATCH] MythMusic: when switching streams, don't reset last track to
 00:00

Rather, preserve his play length to that point in time
---
 mythplugins/mythmusic/mythmusic/musicplayer.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythplugins/mythmusic/mythmusic/musicplayer.cpp b/mythplugins/mythmusic/mythmusic/musicplayer.cpp
index 2942150946a..202701b2794 100644
--- a/mythplugins/mythmusic/mythmusic/musicplayer.cpp
+++ b/mythplugins/mythmusic/mythmusic/musicplayer.cpp
@@ -891,7 +891,7 @@ void MusicPlayer::customEvent(QEvent *event)
         // update the current tracks time in the last played list
         if (m_playMode == PLAYMODE_RADIO)
         {
-            if (!m_playedList.isEmpty())
+            if (!m_playedList.isEmpty() && m_currentTime > 0s)
             {
                 m_playedList.last()->setLength(m_currentTime);
                 // this will update any track lengths displayed on screen
